File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0 or (x % 10 == 0 and x != 0) :
            return False
        rev = 0
        org = x
        while org > 0 :
            d = org % 10
            rev = rev * 10 + d
            org = org / 10
        return rev == x
        

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0 || (x%10==0 && x!=0)) return false;

        int rh = 0;
        while (x>rh)
        {
            rh = rh * 10 + x%10;
            x /= 10;
        }

        return x==rh || x==rh/10;
    }
}